clientconn.go

```diff
@@ -411,6 +433,9 @@ func (cc *ClientConn) Close() error {
 	}
 	cc.state = Shutdown
 	cc.stateCV.Broadcast()
+	if cc.events != nil {
+		cc.events.Finish()
```

Comment:
Ah, I see. Now that we're setting it to nil under the mutex, we need to put all the accesses to events under the same mutex.  I'll make that change.